components/Button/Button.jsx

```diff
@@ -1,91 +1,329 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
+import { shadow, hexToRgba } from '../shared';
+import {
+  components,
+  spacing,
+  colors as defaultColors,
+  baseFontSize as defaultBaseFontSize,
+} from '../shared/theme';
 
 import Icon from '../Icon/Icon';
-import Colors from '../Colors';
-import buttonStyle from './buttonStyle';
-
-const iconSize = ({ size }) => {
-  const sizes = {
-    xsmall: '16px',
-    small: '16px',
-    normal: '24px',
-    large: '24px',
-    xlarge: '24px',
-  };
-
-  return `font-size: ${sizes[size] || sizes.normal};`;
-};
 
 const ButtonIcon = styled(Icon)`
   pointer-events: none;
 `;
 
 const StyledButton = styled.button`
-  ${buttonStyle}
+  align-items: center;
+  display: flex;
+  font-weight: bold;
+  justify-content: center;
+  letter-spacing: 0.2px;
+  border-radius: 4px;
+
+  ${props => `cursor: ${props.disabled ? 'not-allowed' : 'pointer'};`}
+
+  ${({ size, theme: { baseFontSize } }) => {
+    const sizes = {
+      xsmall: `${baseFontSize * 0.75}px`,
+      small: `${baseFontSize * 0.75}px`,
+      medium: `${baseFontSize}px`,
+      large: `${baseFontSize * 1.25}px`,
+      xlarge: `${baseFontSize * 1.5}px`,
+    };
+    return `font-size: ${sizes[size] || sizes.medium};`;
+  }}
+
+  ${({
+    size,
+    theme: {
+      spacing: { large, xlarge, xxlarge, xxxlarge },
+    },
+  }) => {
+    const heights = {
+      xsmall: `${large}px`,
+      small: `${xlarge}px`,
+      medium: `${xxlarge}px`,
+      large: `${xxxlarge}px`,
+    };
+
+    return `height: ${heights[size] || heights.medium};`;
+  }}
+
+  ${({
+    size,
+    theme: {
+      spacing: { small, medium },
+    },
+  }) => {
+    const paddings = {
+      xsmall: `0 ${small}px`,
+      small: `0 ${small}px`,
+      medium: `0 ${medium}px`,
+      large: `0 ${medium}px`,
+      xlarge: `0 ${medium}px`,
+    };
+
+    return `padding: ${paddings[size] || paddings.medium};`;
+  }}
+
+  ${props =>
+    props.center &&
+    `
+    margin-left: auto;
+    margin-right: auto;
+  `}
 
   ${ButtonIcon} {
-    ${iconSize}
+    margin-right: 5px;
```

Comment:
Is `spacing.xxsmall` (4px) not electable in place of this hard coded value?

Reply:
Yes, it is :clap: 

@allyssonsantos